Specs/Scene/ModelExperimental/FeaturePipelineStageSpec.js

```diff
@@ -0,0 +1,251 @@
+import {
+  combine,
+  GltfLoader,
+  FeaturePipelineStage,
+  ShaderBuilder,
+  Resource,
+  ResourceCache,
+} from "../../../Source/Cesium.js";
+import createScene from "../../createScene.js";
+import waitForLoaderProcess from "../../waitForLoaderProcess.js";
+
+describe("Scene/ModelExperimental/FeaturePipelineStage", function () {
+  var buildingsMetadata =
+    "./Data/Models/GltfLoader/BuildingsMetadata/glTF/buildings-metadata.gltf";
+  var microcosm = "./Data/Models/GltfLoader/Microcosm/glTF/microcosm.gltf";
+
+  var scene;
+  var gltfLoaders = [];
+
+  beforeAll(function () {
+    scene = createScene();
+  });
+
+  afterAll(function () {
+    scene.destroyForSpecs();
+  });
+
+  afterEach(function () {
+    var gltfLoadersLength = gltfLoaders.length;
+    for (var i = 0; i < gltfLoadersLength; ++i) {
+      var gltfLoader = gltfLoaders[i];
+      if (!gltfLoader.isDestroyed()) {
+        gltfLoader.destroy();
+      }
+    }
+    gltfLoaders.length = 0;
+    ResourceCache.clearForSpecs();
+  });
+
+  function getOptions(gltfPath, options) {
+    var resource = new Resource({
+      url: gltfPath,
+    });
+
+    return combine(options, {
+      gltfResource: resource,
+      incrementallyLoadTexture: false,
+    });
+  }
+
+  function loadGltf(gltfPath, options) {
+    var gltfLoader = new GltfLoader(getOptions(gltfPath, options));
+    gltfLoaders.push(gltfLoader);
+    gltfLoader.load();
+
+    return waitForLoaderProcess(gltfLoader, scene);
+  }
+
+  function expectUniformMap(uniformMap, expected) {
+    for (var key in expected) {
+      if (expected.hasOwnProperty(key)) {
+        var expectedValue = expected[key];
+        var uniformFunction = uniformMap[key];
+        expect(uniformFunction).toBeDefined();
+        expect(uniformFunction()).toEqual(expectedValue);
+      }
+    }
+  }
+
+  it("sets the feature variables for feature ID attribute", function () {
+    var renderResources = {
+      attributeIndex: 1,
+      pickId: undefined,
+      shaderBuilder: new ShaderBuilder(),
+      model: {
+        _resources: [],
+        featureTable: {
+          _featuresLength: 10,
+          batchTexture: {
+            batchTexture: 0,
+            textureDimensions: {
+              y: 2,
+            },
+            textureStep: 2,
+          },
+        },
+      },
+      runtimePrimitive: {
+        primitive: {},
+      },
+      runtimeNode: {
+        node: {},
+      },
+      uniformMap: {},
+    };
```

Comment:
@ptrgags @sanjeetsuhag - something to start thinking about - the mock objects are starting to get a bit out of hand and could be prone to mistakes as the interfaces change.